cli/cmd/proxy/list/command.go

```diff
@@ -0,0 +1,218 @@
+	"github.com/hashicorp/consul-k8s/cli/common/terminal"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// ListCommand is the command struct for the proxy list command.
+type ListCommand struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+
+	set *flag.Sets
+
+	flagNamespace     string
+	flagAllNamespaces bool
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+// init sets up flags and help text for the command.
+func (c *ListCommand) init() {
+	c.set = flag.NewSets()
+
+	f := c.set.NewSet("Command Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "namespace",
+		Target:  &c.flagNamespace,
+		Default: "default",
+		Usage:   "The namespace to list proxies in.",
+		Aliases: []string{"n"},
+	})
+	f.BoolVar(&flag.BoolVar{
+		Name:    "all-namespaces",
+		Target:  &c.flagAllNamespaces,
+		Default: false,
+		Usage:   "List pods in all namespaces.",
+		Aliases: []string{"A"},
+	})
+
+	f = c.set.NewSet("Global Options")
+	f.StringVar(&flag.StringVar{
+		Name:    "kubeconfig",
+		Aliases: []string{"c"},
+		Target:  &c.flagKubeConfig,
+		Default: "",
+		Usage:   "Set the path to kubeconfig file.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:    "context",
+		Target:  &c.flagKubeContext,
+		Default: "",
+		Usage:   "Set the Kubernetes context to use.",
+	})
+
+	c.help = c.set.Help()
+	c.Init()
+}
+
+// Run executes the list command.
+func (c *ListCommand) Run(args []string) int {
+	c.once.Do(c.init)
+	c.Log.ResetNamed("list")
+	defer common.CloseWithError(c.BaseCommand)
+
+	// Parse the command line flags.
+	if err := c.set.Parse(args); err != nil {
+		c.UI.Output("Error parsing arguments:\n%v", err.Error())
+		return 1
+	}
+
+	// Validate the command line flags.
+	if err := c.validateFlags(); err != nil {
+		c.UI.Output("Invalid argument:\n%v", err.Error())
+		return 1
+	}
+
+	if c.kubernetes == nil {
+		if err := c.initKubernetes(); err != nil {
+			c.UI.Output("Error initializing Kubernetes client", err.Error())
+			return 1
+		}
+	}
+
+	if c.flagAllNamespaces {
+		c.flagNamespace = ""
+	}
+
+	pods, err := c.fetchPods()
+	if err != nil {
+		c.UI.Output("Error fetching pods:\n%v", err, terminal.WithErrorStyle())
+		return 1
+	}
+
+	c.output(pods)
+	return 0
+}
+
+// Help returns a description of the command and how it is used.
+func (c *ListCommand) Help() string {
+	c.once.Do(c.init)
+	return c.Synopsis() + "\n\nUsage: consul-k8s proxy list [flags]\n\n" + c.help
+}
+
+// Synopsis returns a one-line command summary.
+func (c *ListCommand) Synopsis() string {
+	return "List all Pods running Consul proxies."
+}
+
+// validateFlags ensures that the flags passed in by the can be used.
+func (c *ListCommand) validateFlags() error {
+	if len(c.set.Args()) > 0 {
+		return errors.New("should have no non-flag arguments")
+	}
+	if !common.IsValidLabel(c.flagNamespace) {
```

Comment:
`IsValidLabel` appears to be meant for labels, does it make sense to rename this function to something specific about checking for "RFC1123" validity?

Also RFC1123 is a bit tough to find as an end user in google since it is used for many things like DNS and datetime specs.
 Maybe move that error message into the logic of `IsValidLabel()` so we dont have it duplicated elsewhere ?

Two alternatives:
*  we could use this library to replace all of the logic: https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core/validation
* Not duplicate the check logic and let Kubernetes return the error back up the stack

Reply:
Yeah! I will grab that library.